crates/uv-cli/src/lib.rs

```diff
@@ -1586,6 +1586,12 @@ pub struct PipInstallArgs {
     #[arg(long, overrides_with("no_deps"), hide = true)]
     pub deps: bool,
 
+    /// Ignore the package and it's dependencies, only install from the specified dependency group.
+    ///
+    /// May be provided multiple times.
+    #[arg(long, group = "sources", conflicts_with_all = ["requirements", "package", "editable"])]
```

Comment:
Are these conflicts present in the `pip` implementation?

Reply:
(it doesn't really look like it?)

Reply:
I thought `-r` was exclusive to our CLI (I didn't check the others though).

Reply:
Yeah `-r` is exclusive to us, but I'd expect `pip install -e . --group foo` and `pip install anyio --group foo` to work unless they explicitly choose for it not to upstream.

Reply:
(and `-r` should have the same semantics as these other options, I think)

Reply:
One problem is that `uv pip install -r pyproject.toml --extra foo` is the "correct" use of `--extra` but `uv pip install -r pyproject.toml --group foo` would be invalid? Even more confusing, `uv pip install -r foo/pyproject.toml --group bar` would install `bar` from `./pyproject.toml` instead of `foo/pyproject.toml`.

Reply:
Yeah that's part of why I opted to disallow it. The upstream pip semantics are *really* confusing to pair with anything else.

Reply:
I think it's important that we're compatible with pip on whether `--group` can be used with `<package>` and `--editable` though, which introduces all the same complexities.